src/Dropdown.jsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class DisplayContainer extends React.Component {
+  constructor(...args){
+    super(...args);
+    this.handleSelection = this.handleSelection.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.setWrapperRef = this.setWrapperRef.bind(this);
+    this.clicked = this.clicked.bind(this);
+    this.state = {
+      displayValue: 'Good for Day',
+      clicked: false,
+    }
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleClickOutside);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+
+  clicked() {
+    this.setState({
+      clicked: true,
+    });
+  }
+
+  setWrapperRef(node) {
+    this.wrapperRef = node;
+  }
+
+  handleClickOutside(event){
+    if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
+      this.setState({
+        clicked: false,
+      });
+    }
+  }
+
+  handleSelection(item){
+    this.setState({
+      displayValue: item.display
+    })
+  }
+  render(){
+    const listItems = [
+      { display: 'Good for Day', value: 0 },
+      { display: 'Good till Canceled', value: 1 }
+    ];
+
+    let dropdown;
+    if (this.state.clicked) {
+      dropdown = (
+        <div ref={this.setWrapperRef}>
+          <div className='clicked dropdown-width' onClick={this.clicked}>
+            <DropDown options={listItems} value={this.state.displayValue} onClick={this.handleSelection} />
+          </div>
+        </div>
+      );
+    } else {
+      dropdown = (
+        <div ref={this.setWrapperRef}>
+          <div className='notclicked dropdown-width' onClick={this.clicked}>
+            <DropDown options={listItems} value={this.state.displayValue} onClick={this.handleSelection} />
+          </div>
+        </div>
+      );
+    }
+
+    return (
+      <div>
+        {dropdown}
+      </div>
+    );
+  }
+}
+
+class DropDown extends React.Component {
+  static defaultProps = {
+    className: ''
```

Comment:
Great idea to prevent the entire app from breaking due to errors in one component. Have you considered adding a PropType validation as well?